Add _info / _facts module related notes (#169)
ansible-collections/overview: collection_requirements.rst

If your collection is planned to be certified on Automation Hub, please consult with Red Hat Partner Engineering to ensure collection naming compatibility between the community collection on **Galaxy**.

Module naming
-------------

> Modules that only gather information MUST be named ``<something>_info`` or ``<something>_facts``.

Reviewer: ```suggestion
Modules that only gather information MUST be named ``<something>_info``. Modules that only return ``ansible_facts`` are named ``<something>_facts`` and do not return non-facts
```

See `Migrating content to a different collection <https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#migrating-ansible-content-to-a-different-collection>`_ for complete details.

Development conventions
=======================

Besides all the requirements listed in the `Development conventions <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html>`_, be sure:

> * your modules do not return information using the ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -

Reviewer 2: ```suggestion
* your modules do not allow to query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```

Author: committed, thanks!

> * your modules do not allow to query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -

Reviewer 3: ```suggestion
* Your modules do not query information using special ``state`` option values like ``get``, ``list``, ``query``, or ``info`` -
```
